de.dlr.sc.virsat.swtbot.test/src/de/dlr/sc/virsat/swtbot/test/FuncElectricalDiagramTest.java

```diff
@@ -0,0 +1,189 @@
+		SWTBotTreeItem interfaceTypeItem = addElement(InterfaceType.class, conceptFuncElectrical, interfaceTypeCollectionItem);
+		SWTBotTreeItem interfaceEndItem = addElement(InterfaceEnd.class, conceptFuncElectrical, elementConfiguration);
+		
+		dragTreeItemToDiagramEditor(elementConfiguration, diagramEditor, 0, 0);
+		Assert.assertTrue(isEditPartPresentInDiagramEditor(diagramEditor, "ElementConfiguration"));
+		
+		SWTBotGefEditPart editPart = diagramEditor.getEditPart("ElementConfiguration");
+		SWTBotGefEditPart child = editPart.children().get(0);
+
+		dragTreeItemOnToEditPart(interfaceTypeItem, diagramEditor, child);
+				
+		String text = interfaceEndItem.expand().getNode(0).getText();
+		Assert.assertTrue(text.equals("type -> InterfaceType"));		
+	}
+	
+	@Test
+	public void removeInterfaceDiagramElementUndoRedoTest() {
+		addElement(Interface.class, conceptFuncElectrical, elementConfiguration);
+		dragTreeItemToDiagramEditor(elementConfiguration, diagramEditor);
+		Assert.assertTrue(isEditPartPresentInDiagramEditor(diagramEditor, "ElementConfiguration"));
+		
+		removeEditPartInDiagramEditor(diagramEditor, "ElementConfiguration");
+		Assert.assertFalse(isEditPartPresentInDiagramEditor(diagramEditor, "ElementConfiguration"));		
+		
+		undoCommand();
+		Assert.assertTrue(isEditPartPresentInDiagramEditor(diagramEditor, "ElementConfiguration"));
+		redoCommand();
+		Assert.assertFalse(isEditPartPresentInDiagramEditor(diagramEditor, "ElementConfiguration"));
+	}
+	
+	@Test
+	public void deleteInterfaceDiagramElementUndoRedoTest() {
+		addElement(Interface.class, conceptFuncElectrical, elementConfiguration);
+		dragTreeItemToDiagramEditor(elementConfiguration, diagramEditor);
+		Assert.assertTrue(isEditPartPresentInDiagramEditor(diagramEditor, "ElementConfiguration"));
+		
+		String elementConfigurationName = elementConfiguration.getText();
+		
+		deleteEditPartInDiagramEditor(diagramEditor, "ElementConfiguration");
+		Assert.assertFalse(isEditPartPresentInDiagramEditor(diagramEditor, "ElementConfiguration"));
+		Assert.assertFalse(isTreeItemPresentInTreeView(elementConfiguration));
+		
+		diagramEditor.setFocus();
+		
+		undoCommand();
+		Assert.assertTrue(isEditPartPresentInDiagramEditor(diagramEditor, "ElementConfiguration"));
+		
+		SWTBotTreeItem elementConfigurationNode = configurationTree.getNode(elementConfigurationName);
+		Assert.assertNotNull(elementConfigurationNode);
+		
+		diagramEditor.setFocus();
+
+		redoCommand();
+		Assert.assertFalse(isEditPartPresentInDiagramEditor(diagramEditor, "ElementConfiguration"));
+		Assert.assertFalse(isTreeItemPresentInTreeView(elementConfigurationNode));
+	}
+	
+	@Test
+	public void connectInterfaceEndsTest() {		
+		addElement(Interface.class, conceptFuncElectrical, elementConfiguration);
+		
+		dragTreeItemToDiagramEditor(elementConfiguration, diagramEditor, DRAG1_COORDINATES_X, DRAG1_COORDINATES_Y);
+		SWTBotGefEditPart swtBotGefEditPart1 = diagramEditor.selectedEditParts().get(0);		
```

Comment:
Should have a proper name instead of just genericaly being called editPart1.
Are they athe structural element instances in the diagram? This should be visible from their name.